spec/scenarios/customers/create_partner_and_run_billing_spec.rb

```diff
@@ -0,0 +1,191 @@
+    travel_to(apr24) do
+      create(
+        :package_charge,
+        plan: plan,
+        billable_metric: metric,
+        pay_in_advance: false,
+        prorated: false,
+        invoiceable: true,
+        properties: {
+          amount: '2',
+          free_units: 1000,
+          package_size: 1000
+        }
+      )
+
+      customers.each do |customer|
+        create_subscription(
+          {
+            external_customer_id: customer.external_id,
+            external_id: customer.external_id,
+            plan_code: plan.code
+          }
+        )
+      end
+      create_subscription(
+        {
+          external_customer_id: partner.external_id,
+          external_id: partner.external_id,
+          plan_code: plan.code
+        }
+      )
+    end
+
+    ### 25 Apr: Ingest events for Plan 1.
+    apr24 = Time.zone.parse('2024-04-24')
+    travel_to(apr24) do
+      plan.subscriptions.each do |subscription|
+        create_event(
+          params.merge(
+            external_subscription_id: subscription.external_id
+          )
+        )
+      end
+      perform_all_enqueued_jobs
+    end
+
+    # May 1st: Billing run; check invoice numbering
+    may1 = Time.zone.parse('2024-05-1')
+    travel_to(may1) do
+      organization.update(created_at: 1.month.ago)
+      perform_billing
+      expect(organization.invoices.count).to eq(3)
+      expect(partner.invoices.count).to eq(1)
+
+      partner_invoice = partner.invoices.first
+      expect(partner_invoice.self_billed).to eq(true)
+      expect(partner_invoice.number).to eq("#{organization.document_number_prefix}-001-001")
+
+      organization_invoices = customers.map(&:invoices).flatten
+      expect(organization_invoices.map(&:self_billed).uniq).to eq([false])
+      expect(organization_invoices.map do |inv|
+        inv.number.gsub("#{organization.document_number_prefix}-202405-", '')
+      end.uniq.sort).to eq(['001', '002'])
+    end
+
+    # June 1st: Billing run; check invoice numbering
+    june1 = Time.zone.parse('2024-06-1')
+    travel_to(june1) do
+      perform_billing
+      expect(organization.invoices.count).to eq(6)
+      expect(partner.invoices.count).to eq(2)
+
+      partner_invoice = partner.invoices.where(created_at: june1).first
+      expect(partner_invoice.self_billed).to eq(true)
+      expect(partner_invoice.number).to eq("#{organization.document_number_prefix}-001-002")
+
+      organization_invoices = customers.map { |c| c.invoices.where(created_at: june1) }.flatten
+      expect(organization_invoices.map(&:self_billed).uniq).to eq([false])
+      expect(organization_invoices.map do |inv|
+        inv.number.gsub("#{organization.document_number_prefix}-202406-", '')
+      end.uniq.sort).to eq(['003', '004'])
+    end
+    update_overdue_balance
+
+    # check payments
+    # partner_invoice = partner.invoices.where(created_at: may1).first
+    expect(partner.invoices.map(&:payments).flatten.count).to be(0)
```

Comment:
or... whatever you like the most

```suggestion
    expect(partner.invoices.map(&:payments).flatten).to be_empty
```